KIALI-2560 Add expanded mode for overview cards

jotak opened the pull request:
See https://github.com/kiali/kiali-design/issues/57

- Fetch metrics in overview when mode is expanded
- Add toggle button to switch expanded/compact
- Horizontal bar chart (for health) and sparkline (for RPS)

JIRA: https://issues.jboss.org/browse/KIALI-2560

jotak commented:
Not ready to merge (there's a couple of things I want to improve ; for instance when no data is available (N/A))

But early feedback is welcome.
cc @serenamarie125 @abonas @lucasponce 

![sshot](https://screenshotscdn.firefoxusercontent.com/images/31c984b4-4387-4bfa-b261-799a2df2a3f3.png)

serenamarie125 commented:
@jotak thanks for sharing!  originally i thought left aligning made sense, but now wondering if the content above and below the bar chart should be horizontally center aligned instead.  What do you think ?  
<img width="637" alt="Screen Shot 2019-03-21 at 3 47 30 PM" src="https://user-images.githubusercontent.com/12504403/54760604-b3114b00-4bf0-11e9-9643-239a5da6df17.png">

jotak commented:
New screenshot with centered text and adding a title for the sparkline:

![2](https://screenshotscdn.firefoxusercontent.com/images/b80e0b06-7688-4664-b9d0-31eb48e1a735.png)

(I merged "Traffic" and "last X minutes" on a single line, tell me if it's not ok)

About left-aligned versus centered, I'm fine with both.

hhovsepy commented:
Tested from functionality pov, Compact/Expand, Time Range, Overview types.

xeviknal commented on src/pages/Overview/OverviewPage.tsx:
```diff
@@ -299,6 +345,47 @@ class OverviewPage extends React.Component<OverviewProps, State> {
     );
   }
+
+  renderStatuses(ns: NamespaceInfo): JSX.Element {
```
What about having an smaller component with those Statuses? Is it too much fragmentation? I see this component is growing to 400L. I think is a smell that things should be moved. wdyt?

xeviknal commented on src/pages/Overview/OverviewPage.tsx:
```diff
@@ -299,6 +345,47 @@ class OverviewPage extends React.Component<OverviewProps, State> {
+    return <div style={{ height: 70 }} />;
+  }
+
+  renderCardLinks(ns: NamespaceInfo): JSX.Element {
```
It might work also here. An smaller component rendering only the links. wdyt?

jotak commented:
@xeviknal I've split Cards Links to another component, and moved a static function to the helper class.
But no change about Statuses (as there's already a component).
